File: js/src/types/attestation-gateway.ts
import type { AttestationEvidence } from './attestation-common';
import type { OhttpAttestation } from './ohttp';

/** Raw NEAR AI Cloud Gateway evidence. */
export type GatewayAttestation = AttestationEvidence & {
  /** Gateway-reported TLS SPKI fingerprint when the evidence request enables TLS binding. */
  spkiFingerprint?: string;
  reportedQuoteData: string;
  /** Signed OHTTP configuration from the report envelope; verify against this Gateway's authenticated signer. */
  ohttpAttestation?: OhttpAttestation;
};

File: py/src/verifiable_ai_sdk/types/attestation_gateway.py
"""Raw Gateway attestation evidence returned by NEAR AI Cloud."""

from __future__ import annotations

from dataclasses import dataclass

from .attestation_common import AttestationEvidence


@dataclass(frozen=True, kw_only=True)
class GatewayAttestation(AttestationEvidence):
    """A NEAR AI Cloud Gateway TEE attestation."""

    #: Gateway-reported SPKI fingerprint, present when the evidence request
    #: requested it.
    spki_fingerprint: str | None = None
    reported_quote_data: str
